### PR TITLE
Replaced Text with String in ArgumentParser and Csv* classes

### PR DESCRIPTION
**Pull request by marcusnaslund**
(Major) part of #1724 - removing use of `Text`.

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Peer review @sebastianbaginski ?

File: source/cli/ArgumentParser.ooc [hunk 1]
```diff
@@ -48,68 +49,68 @@ ArgumentParser: class {
...
-				current = current slice(1)
-				for (j in 0 .. current count)
-					arguments add(current slice(j, 1))
-			}
+			current := input[i]
+			if (current startsWith("--"))
+				arguments add(current substring(2))
+			else if (current startsWith("-"))
+				for (j in 1 .. current size)
+					arguments add(current[j] as String)
```
Comment (sebastianbaginski) [inline source/cli/ArgumentParser.ooc]: Why does it work ? Bracket operator in String class returns a Char : `operator [] (string: String, index: Int) -> Char { string _buffer [index] }` . Confusing :/

Comment (sebastianbaginski) [inline source/cli/ArgumentParser.ooc]: Ah ok, Char has `as String` operator. Nevermind.

Comment (marcusnaslund) [inline source/cli/ArgumentParser.ooc]: Because you can overload the `as` operator. So `current[j] as String` actually creates a new `String` of length 1 with `current[j]` as the only character.

File: source/io/CsvReader.ooc [hunk 2]
```diff
@@ -59,38 +62,35 @@ CsvReader: class extends Iterator<VectorList<Text>> {
...
-						textBuilder append(readCharacter)
+						stringBuilder add(readCharacter)
 				}
 			}
-			result add(textBuilder toText())
-			textBuilder free()
+			result add(stringBuilder toString())
+			for (i in 0 .. stringBuilder count)
+				stringBuilder[i] free()
+			stringBuilder free()
```
Comment (sebastianbaginski) [inline source/io/CsvReader.ooc]: `stringBuilder freeAll()` ?

File: source/io/CsvReader.ooc [hunk 3]
```diff
@@ -59,38 +62,35 @@ CsvReader: class extends Iterator<VectorList<Text>> {
...
+			stringBuilder add(rowData[position])
 			position += 1
 		}
 		position += 1
-		result = textBuilder toText()
-		textBuilder free()
+		result = stringBuilder toString()
+		for (i in 0 .. stringBuilder count)
+			stringBuilder[i] free()
+		stringBuilder free()
```
Comment (sebastianbaginski) [inline source/io/CsvReader.ooc]: As above.

File: source/io/CsvReader.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/io/CsvReader.ooc]: Or, the `StringBuilder` constructor should take an argument of whether it should free the content or not. That will be another PR though.

**Comment by sebastianbaginski** [comment]
Ok then.

**Comment by marcusnaslund** [comment]
Thanks. Noted that this is green, taking it off until corresponding PR is ready.

File: source/io/CsvReader.ooc (hunk 2 above)
Comment (marcusnaslund) [inline source/io/CsvReader.ooc]: It was way too easy to fix ( #1825 ) so I will be incorporating it here anyway.

**Comment by marcusnaslund** [comment]
`./test.sh && ./tools/memcheck.sh` shows 0 new leaks, yay.

**Comment by fredrikbryntesson** [comment]
OK
